app/src/main/java/com/google/android/sambadocumentsprovider/provider/SambaDocumentsProvider.java

```diff
@@ -316,6 +344,16 @@ public void onTaskFinished(
     }
   }
 
+  private DocumentMetadata getMetadataFromCacheResult(CacheResult cacheResult, Uri documentUri) {
+    if (cacheResult.getState() == CacheResult.CACHE_MISS) {
+      final DocumentMetadata metadata = DocumentMetadata.createShare(documentUri);
```

Comment:
This feels like a hack, because it's not clear why cache miss leads to creating a share metadata.

Suggest keeping the old change and adding the share document metadata to the cache if cache doesn't have it.